Yosemite/Yosemite/Actions/CardPresentPaymentAction.swift

```diff
@@ -21,4 +21,11 @@ public enum CardPresentPaymentAction: Action {
                         parameters: PaymentParameters,
                         onCardReaderMessage: (CardReaderEvent) -> Void,
                         onCompletion: (Result<Void, Error>) -> Void )
+
+    case checkForUpdate(onData: (Result<CardReaderSoftwareUpdate, Error>) -> Void,
```

Comment:
This is tightly coupled to [the way the Stripe Terminal SDK checks for updates](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPTerminal.html#/c:objc(cs)SCPTerminal(im)checkForUpdate:). In a nutshell, when the completion block of the method that checks for updates will get called when an update is found, or when there is an error, but also if there are not new updates found. 

It kinds of make sense to have two callback in Yosemite though, one for "hey, there is one update available" and another one for "I'm done, there is nothing available". It feels a bit more declarative than checking an optional for nil